applications/ShapeOptimizationApplication/python_scripts/analyzer_internal.py

```diff
@@ -27,9 +29,26 @@
 except ImportError:
     convdiff_response_factory = None
 
-
 import time as timer
 
+class IterationScope:
+    def __init__(self, output_folder_name, response_id, iteration_number):
+        if (output_folder_name != None):
+            self.currentPath = pathlib.Path.cwd()
+            output_path = pathlib.Path(output_folder_name)
+            response_text = "response_{:}/{:d}".format(response_id, iteration_number)
+            self.scope = output_path / response_text
+
+    def __enter__(self):
+        if (hasattr(self, "scope")):
+            self.scope.mkdir(parents=True, exist_ok=True)
+            os.chdir(str(self.scope))
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if (hasattr(self, "currentPath")):

```

Comment:
same

Reply:
not exactly. I only set the scope if folder structure is required only.

Reply:
oh i wanted to delete this comment. please ignore

Reply:
removed `hasattr`